File: CQ_SDK/StatusEvent.cpp
#include "StatusEvent.h"

#include "Unpack.h"
#include "tool.h"
#include "event.h"

void CQ::StatusEvent::color_green() { color = Ðü¸¡´°_ÂÌ; }
void CQ::StatusEvent::color_orange() { color = Ðü¸¡´°_³È; }
void CQ::StatusEvent::color_red() { color = Ðü¸¡´°_ºì; }
void CQ::StatusEvent::color_crimson() { color = Ðü¸¡´°_Éîºì; }
void CQ::StatusEvent::color_black() { color = Ðü¸¡´°_ºÚ; }
void CQ::StatusEvent::color_gray() { color = Ðü¸¡´°_»Ò; }

std::string CQ::statusEVEreturn(StatusEvent & e)
{
	Unpack pack;
	std::string _ret = pack.add(e.data).add(e.dataf).add(e.color).getAll();
	_ret = base64_encode(_ret);
	return _ret;
}

File: CQ_HELP/src/事件/StatusEvent.java
/*
 * Copyright (C) 2018 admin
 *
 * This program is free software: you can redistribute it and/or modify
 * it under the terms of the GNU General Public License as published by
 * the Free Software Foundation, either version 3 of the License, or
 * (at your option) any later version.
 *
 * This program is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
 * GNU General Public License for more details.
 *
 * You should have received a copy of the GNU General Public License
 * along with this program.  If not, see <http://www.gnu.org/licenses/>.
 */
package 事件;

import c.string;

/**
 * 悬浮窗事件.
 * @author MikuPy2001
 */
public abstract class StatusEvent {

    //数据
    public string data;
    //数据单位
    public string dataf;
    // 1 : 绿
    // 2 : 橙
    // 3 : 红
    // 4 : 深红
    // 5 : 黑
    // 6 : 灰
    public int color;

    /**
     * 1 : 绿
     */
    public abstract void color_green();

    /**
     * 2 : 橙
     */
    public abstract void color_orange();

    /**
     * 3 : 红
     */
    public abstract void color_red();

    /**
     * 4 : 深红
     */
    public abstract void color_crimson();

    /**
     * 5 : 黑
     */
    public abstract void color_black();

    /**
     * 6 : 灰
     */
    public abstract void color_gray();
}
